gcc/testsuite/rust/compile/name_resolution4.rs

```diff
@@ -0,0 +1,13 @@
+trait Foo {
+    fn foo(&self) {} // { dg-warning "unused name" }
+}
+
+struct Bar;
+
+pub fn bar() {
+    impl Foo for Bar {}
+}
+
+fn main() {
+    Bar.foo();
```

Comment:
name-resolution does not resolve the method it only resolves the expression path of Bar.

Reply:
right, and that's something we need to handle with the new name resolution too (deferring to type-checking or however we call it at the moment)